server/src/operators/chunk_operator.rs

```diff
@@ -1438,22 +1439,28 @@ pub fn get_highlights_with_exact_match(
         engine.insert(i, x);
     });
 
+    let mut query_parts_used = HashSet::new();
+
     let mut matched_idxs = split_content
         .iter()
         .enumerate()
-        .filter_map(
-            |(i, split)| match query_parts_split_by_stop_words.contains(split) {
-                true => Some((i, split.clone())),
+        .filter_map(|(i, split)| {
+            match query_parts_split_by_stop_words.contains(&split.to_lowercase()) {
+                true => {
+                    query_parts_used.insert(split.clone());
+                    Some((i, split.clone()))
+                },
```

Comment:
This solves an actual bug, this should remain mostly because we were incorrectly not lowercasing our query and splits when comparing previously